pwa/src/components/DiagnosisForm.tsx

```diff
@@ -0,0 +1,109 @@
+import React, { ReactElement } from 'react'
+import { List, Checkbox, InputItem, WhiteSpace } from 'antd-mobile'
+import '../scss/DiagnosisForm.scss'
+import * as data from './DiagnosisFields.json'
+
+const CheckboxItem = Checkbox.CheckboxItem
+
+/*
+
+Diagnosis Fields are stored in DiagnosisFields.json in the format:
+
+{
+    "diagnoses" : [
+        {
+            "name": "Name",
+            "id": "id",
+            "text-input-title": "Title",
+            "text-input-type": "type" (i.e. number)
+            "group-title": "Group Title",
+            "group": [
+                { "name": "Name1", "id": "id1" },
+                { "name": "Name2", "id": "id2" }
+            ]
+        },
+        {...}
+    ]
+}
+*/
+
+const diagnosisFields = data.diagnoses
+
+export default function DiagnosisForm(): ReactElement {
+    return (
+        <div id="diagnosis-form-wrapper" data-testid="diagnosis-form-wrapper">
+            {diagnosisFields.map((field) => {
+                const formElements: ReactElement[] = []
+
+                if (field.name != null) {
+                    formElements.push(
+                        <CheckboxItem
+                            id={field.id}
+                            data-testid={field.id}
+                            key={field.id}
+                        >
+                            {field.name}
+                        </CheckboxItem>,
+                    )
+                }
+
+                if (
+                    field['text-input-title'] != null &&
+                    field['text-input-id'] != null
+                ) {
+                    formElements.push(
+                        <List renderHeader={() => field['text-input-title']}>
+                            <InputItem
+                                id={field['text-input-id']}
+                                data-testid={field['text-input-id']}
+                                type={
+                                    field['text-input-type'] == 'number'
+                                        ? 'number'
+                                        : 'text'
+                                }
+                                maxLength={
+                                    field['text-input-id'] == 'pregnancy-weeks'
+                                        ? 2
+                                        : null
+                                }
```

Comment:
I'm not absolutely thrilled about hardcoding this id here, but it is the only number input in the form so there shouldn't be too many other cases. I am open to suggestions for a better way to do this.

Reply:
How about this @meghanmc ? Putting a max-limit on the JSON object.

[image](https://user-images.githubusercontent.com/33632286/80436935-ef1fe080-88c5-11ea-8737-93fbeb905293.png)

I'm gonna commit it now just so I can continue to browse around, but this can be reverted if needed 👍

[Commit](https://github.com/Haiti-Distributed-Healthcare-System/hdhs/pull/84/commits/0b8f032476aceda60b087470575bccd6a37397e0)

Reply:
I like this solution, thanks